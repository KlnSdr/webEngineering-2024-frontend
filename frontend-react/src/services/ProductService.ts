import {NeededProduct, Product} from "../types/Products";

class ProductsService {
  private static backendURL: string =
    process.env.REACT_APP_BACKEND_URL || "http://localhost:13000";

  public static getAll(): Promise<Product[]> {
    return new Promise((resolve, reject) => {
        fetch(`${this.backendURL}/products`).then((response: Response) => {
            if (!response.ok) {
                throw new Error("Failed to load products.");
            }
            return response.json();
        }).then((data: Product[]) => {
            console.log(data);
            resolve(data);
        }).catch((reason: any) => {
            reject(reason);
        });
    });
  }
  
  //get Id from product
    public static getProductIdByName(productName: string): Promise<NeededProduct | null> {
        return this.getAll().then(products => {
            const product = products.find(prod => prod.name.toLowerCase() === productName.toLowerCase());
            return product ? { id: product.id, productName: product.name, amount: 0 } : null;
        });
    }

}

export { ProductsService };
