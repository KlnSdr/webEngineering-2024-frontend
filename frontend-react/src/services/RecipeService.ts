import { CreateRecipe } from "../types/Recipes";
import { Recipe } from "../types/Recipes";
import {NeededProduct} from "../types/Products";

class RecipeService {
  private static backendURL: string =
    process.env.REACT_APP_BACKEND_URL || "http://localhost:13000";

  public static save(recipe: CreateRecipe): Promise<void> {
    return new Promise((resolve, reject) => {
      fetch(`${this.backendURL}/recipes`, {
        method: "POST",
        credentials: "include",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(recipe),
      })
        .then((response: Response) => {
          if (!response.ok) {
            throw new Error("Could not save recipe.");
          }
          return response.json();
        })
        .then((data: any) => {
          console.log(data);
          resolve();
        })
        .catch((reason: any) => {
          reject(reason);
        });
    });
  }
  public static getAll1(): Promise<CreateRecipe[]> {
    return new Promise((resolve, reject) => {
      resolve([
        {
          title: "Käsesoße 1",
          image: null,
          description: "Description 1",
          products: [],
        },
        {
          title: "Käsesoße 2",
          image: null,
          description: "Description 2",
          products: [],
        },
        {
          title: "Käsesosse 3",
          image: null,
          description: "Description 3",
          products: [],
        },
        {
          title: "Käsesosse 4",
          image: null,
          description: "Description 3",
          products: [],
        },
      ]);
    });
  }

  public static getAll():Promise<Recipe[]> {
    return new Promise((resolve, reject) => {
      fetch(`${this.backendURL}/recipes`).then((response: Response) => {
          if (!response.ok) {
            throw new Error("Failed to load recipes.");
          }
          return response.json();
        }).then((data: Recipe[]) => {
          resolve(data);
        })
        .catch((reason: any) => {
          reject(reason);
        });
    });
  }

    public static getAll(): Promise<CreateRecipe[]> {
        return new Promise((resolve, reject) => {
            resolve([
                {
                    title: "Käsesoße 1",
                    image: null,
                    description: "A delicious cheese sauce.",
                    products: [
                        { id: 7, productName: "Cheese", amount: 1 },
                        { id: 4, productName: "Milk", amount: 1 },
                        { id: 6, productName: "Butter", amount: 1 },
                    ],
                },
                {
                    title: "Käsesoße 2",
                    image: null,
                    description: "Cheese sauce with salt.",
                    products: [
                        { id: 7, productName: "Cheese", amount: 2 },
                        { id: 10, productName: "Salt", amount: 1 },
                        { id: 4, productName: "Milk", amount: 2 },
                    ],
                },
                {
                    title: "Käsesoße 3",
                    image: null,
                    description: "A spicy cheese sauce.",
                    products: [
                        { id: 7, productName: "Cheese", amount: 2 },
                        { id: 11, productName: "Pepper", amount: 1 },
                        { id: 2, productName: "Milk", amount: 1 },
                    ],
                },
                {
                    title: "Käsesoße 4",
                    image: null,
                    description: "A mild cheese sauce.",
                    products: [
                        { id: 7, productName: "Cheese", amount: 3 },
                        { id: 2, productName: "Milk", amount: 2 }
                    ],
                },
            ]);
        });
    }


    public static searchRecipesByProducts(neededProducts: NeededProduct[]): Promise<CreateRecipe[]> {
        return new Promise((resolve, reject) => {
            this.getAll()
                .then((recipes) => {

                    const staticRecipes: CreateRecipe[] = [
                        {
                            title: "Käsesoße 1",
                            image: null,
                            description: "A delicious cheese sauce.",
                            products: [
                                { id: 7, productName: "Cheese", amount: 1 },
                                { id: 4, productName: "Milk", amount: 1 },
                                { id: 6, productName: "Butter", amount: 1 },
                            ],
                        },
                        {
                            title: "Käsesoße 2",
                            image: null,
                            description: "Cheese sauce with salt.",
                            products: [
                                { id: 7, productName: "Cheese", amount: 2 },
                                { id: 10, productName: "Salt", amount: 1 },
                                { id: 4, productName: "Milk", amount: 2 },
                            ],
                        },
                        {
                            title: "Käsesoße 3",
                            image: null,
                            description: "A spicy cheese sauce.",
                            products: [
                                { id: 7, productName: "Cheese", amount: 2 },
                                { id: 11, productName: "Pepper", amount: 1 },
                                { id: 2, productName: "Milk", amount: 1 },
                            ],
                        },
                        {
                            title: "Käsesoße 4",
                            image: null,
                            description: "A mild cheese sauce.",
                            products: [
                                { id: 7, productName: "Cheese", amount: 3 },
                                { id: 2, productName: "Milk", amount: 2 }
                            ],
                        },
                    ];

                    resolve(staticRecipes);
                })
                .catch((reason: any) => {
                    reject(reason);
                });
        });
    }

}

export { RecipeService };
