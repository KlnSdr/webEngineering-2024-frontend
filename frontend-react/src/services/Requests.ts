// just one more abstraction bro :D

/**
 * Sends a request to the specified path with the given options.
 * @param {string} path - The URL path to send the request to.
 * @param {Object} [options={}] - The options to use for the request.
 * @returns {Promise<Response>} A promise that resolves to the response of the request.
 */
function request(path: string, options: any = {}): Promise<Response> {
    return fetch(path, options);
}

/**
 * Sends an authorized request to the specified path with the given options.
 * Adds an Authorization header with a Bearer token if available.
 * @param {string} path - The URL path to send the request to.
 * @param {Object} [options={}] - The options to use for the request.
 * @returns {Promise<Response>} A promise that resolves to the response of the request.
 */
function authorizedRequest(path: string, options: any = {}): Promise<Response> {
    const token = localStorage.getItem("token");
    if (token) {
        options.headers = {
            ...options.headers,
            Authorization: `Bearer ${token}`,
        };
    }
    return request(path, options);
}

export { request, authorizedRequest };