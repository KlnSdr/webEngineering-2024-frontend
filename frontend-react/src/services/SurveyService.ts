import {CreateSurvey, Survey} from "../types/Surveys";;

class SurveyService {
    private static backendURL: string =
        process.env.REACT_APP_BACKEND_URL || "http://localhost:13000";

    public static getSurveysByUserId(userId: string): Promise<any[]> {
        return new Promise((resolve, reject) => {
            fetch(`${this.backendURL}/surveys/user/${userId}`).then((response: Response) => {
                if (!response.ok) {
                    throw new Error("Failed to load surveys.");
                }
                return response.json();
            }).then((data: Survey[]) => {
                resolve(data);
            }).catch((reason: any) => {
                reject(reason);
            });
        });
    }

    public static getSurveyById(surveyId: string | undefined): Promise<Survey> {
        return new Promise((resolve, reject) => {
            fetch(`${this.backendURL}/surveys/${surveyId}`, {credentials: "include"}).then((response: Response) => {
                if (!response.ok) {
                    throw new Error("Failed to load survey.");
                }
                return response.json();
            }).then((data: Survey) => {
                console.log(data);
                resolve(data);
            }).catch((reason: any) => {
                reject(reason);
            });
        });
    }
    public static voteSurvey(surveyId: number, recipeId: number): Promise<Survey>{
        return new Promise((resolve, reject) =>{
            fetch(`${this.backendURL}/surveys/${surveyId}/vote/${recipeId}`, {
                method: "PUT",
                credentials: "include",
                headers: {
                    "Content-Type": "application/json",
                },
            }).then((response: Response) => {
                if (!response.ok) {
                    throw new Error("Failed to vote.");
                }
                return response.json();
            }).then((data: Survey) => {
                resolve(data);
            }).catch((reason: any) => {
                reject(reason);
            });
        });
    }

    public static createSurvey(survey: CreateSurvey): Promise<void> {
        const surveyToSend = {
            title: survey.title,
            options: survey.options.map((recipe: any) => `/recipes/${recipe.id}`),
        };

        return new Promise((resolve, reject) => {
            fetch(`${this.backendURL}/surveys`, {
                method: "POST",
                credentials: "include",
                headers: {
                    "Content-Type": "application/json",
                },
                body: JSON.stringify(surveyToSend),
            }).then((response: Response) => {
                if (!response.ok) {
                    throw new Error("Could not save survey.");
                }
                return response.json();
            }).then((data: any) => {
                console.log(data);
                resolve();
            }).catch((reason: any) => {
                reject(reason);
            });
        });
    }
}

export default SurveyService;