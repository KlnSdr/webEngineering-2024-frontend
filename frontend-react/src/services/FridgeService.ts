import { NeededProduct } from "../types/Products";
import {ProductsService} from "./ProductService";


class FridgeService {
    private static backendURL: string =
        process.env.REACT_APP_BACKEND_URL || "http://localhost:13000";

    public static getFridgeContent(userId: number): Promise<NeededProduct[]> {
        return new Promise((resolve, reject) => {
            fetch(`${this.backendURL}/fridge/${userId}`, {
                method: "GET",
                credentials: "include",
            })
                .then((response) => {
                    if (!response.ok) {
                        throw new Error("Failed to load fridge content.");
                    }
                    return response.json();
                })
                .then(async (data: { userUri: string; products: Record<string, number> }) => {
                    const products: NeededProduct[] = [];

                    // Fetch product IDs for each product name
                    for (const [productName, amount] of Object.entries(data.products)) {
                        const productId = await ProductsService.getProductIdByName(productName);
                        if (productId !== null) {
                            products.push({
                                id: productId.id,
                                productName: productId.productName,
                                amount,
                            });
                        } else {
                            console.warn(`Product ID not found for ${productName}`);
                        }
                    }

                    resolve(products);
                })
                .catch((reason: any) => {
                    reject(reason);
                });
        });
    }

    public static updateFridgeContent(userId: number, products: NeededProduct[]): Promise<void> {
        return new Promise((resolve, reject) => {
            // Convert the products array to a map
            const productsMap: Record<string, number> = {};
            products.forEach((product) => {
                productsMap[product.productName] = product.amount;
            });

            fetch(`${this.backendURL}/fridge/${userId}`, {
                method: "PUT",
                credentials: "include",
                headers: {
                    "Content-Type": "application/json",
                },
                body: JSON.stringify({ userUri: `/users/${userId}`, products: productsMap }),
            })
                .then((response) => {
                    if (!response.ok) {
                        throw new Error("Failed to update fridge content.");
                    }
                    resolve();
                })
                .catch((reason: any) => {
                    reject(reason);
                });
        });
    }

}

export { FridgeService };
