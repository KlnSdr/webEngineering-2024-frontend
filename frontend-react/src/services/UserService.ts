import {CurrentUser, User} from "../types/Users";
import {jwtDecode} from "jwt-decode";
import {request} from "./Requests";

/**
 * Service class for handling user-related operations.
 */
class UserService {
  /**
   * The backend URL for the user service.
   * @type {string}
   * @private
   */
  private static backendURL: string =
    process.env.REACT_APP_BACKEND_URL || "http://localhost:13000";

  /**
   * Cached user information.
   * @type {CurrentUser | null}
   * @private
   */
  private static userInfo: CurrentUser | null = null;

  /**
   * Logs out the current user by removing the token and redirecting to the logout URL.
   */
  static doLogout() {
    localStorage.removeItem("token");
    window.location.assign(`${UserService.backendURL}/users/logout`);
  }

  /**
   * Initiates the login process by redirecting to the OAuth2 authorization URL.
   */
  static doLogin() {
    window.location.assign(
      `${UserService.backendURL}/oauth2/authorization/gitlab`
    );
  }

  /**
   * Retrieves the current user's information.
   * @returns {Promise<any>} A promise that resolves to the current user's information.
   */
  static getUserInfo(): Promise<any> {
    return new Promise<any>((resolve, reject) => {
      if (UserService.userInfo !== null) {
        resolve(UserService.userInfo);
        return;
      }

      if (localStorage.getItem("token")) {
          this.loadUserInfo(localStorage.getItem("token")!);
      }

      resolve(UserService.userInfo);
    });
  }

  /**
   * Retrieves a user by their URI.
   * @param {string} uri - The URI of the user.
   * @returns {Promise<User>} A promise that resolves to the user.
   */
  static getUser(uri: string): Promise<User> {
      return new Promise((resolve, reject) => {
          request(`${this.backendURL}${uri}`, {})
          .then(response => {
              if (!response.ok) {
                  throw new Error("could not get user with uri " + uri);
              }
              return response.json();
          })
          .then(resolve)
          .catch(_ => {reject()});
      });
  }

  /**
   * Checks if the user is logged in.
   * @returns {Promise<boolean>} A promise that resolves to a boolean indicating if the user is logged in.
   */
  static isLoggedIn(): Promise<boolean> {
    return new Promise<boolean>((resolve) => {
      UserService.init().then((_) => resolve(UserService.userInfo !== null));
    });
  }

  /**
   * Loads the user information from the given token.
   * @param {string} token - The JWT token.
   * @private
   */
  static loadUserInfo(token: string) {
    UserService.userInfo = jwtDecode<CurrentUser>(token);
  }

  /**
   * Initializes the user service by loading the user information.
   * @returns {Promise<void>} A promise that resolves when the initialization is complete.
   * @private
   */
  private static init(): Promise<void> {
    return new Promise((resolve) => {
      UserService.getUserInfo()
        .then((_) => resolve())
        .catch((_) => resolve());
    });
  }
}

export { UserService };