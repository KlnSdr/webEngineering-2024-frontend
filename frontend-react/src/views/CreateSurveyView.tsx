import React, { useEffect, useState } from "react";
import "../style/CreateRecipeView.css";
import { UserService } from "../services/UserService";
import { Link, useNavigate } from "react-router-dom";
import LabelInput from "../components/LabelInput";
import Button from "react-bootstrap/Button";
import Stack from "react-bootstrap/Stack";
import RecipeSearch from "../components/RecipeSearch";
import Heading from "../components/Heading";
import ImageArea from "../components/ImageArea";
import MyRecipeBar from "../components/MyRecipeBar";
import { CreateSurvey, Survey, UpdateSurvey } from "../types/Surveys";
import { Recipe } from "../types/Recipes";
import Alert from "react-bootstrap/Alert";
import SurveyService from "../services/SurveyService";

/**
 * Component for creating or editing a survey.
 * @param {Object} props - The component props.
 * @param {UpdateSurvey | null} props.survey - The survey to edit, or null to create a new survey.
 * @returns {JSX.Element} The rendered component.
 */
function CreateSurveyView({ survey }: { survey: UpdateSurvey | null }): JSX.Element {

    /**
     * The initial state for a new survey.
     * @type {CreateSurvey}
     */
    const emptySurvey: CreateSurvey = {
        title: "",
        options: []
    };

    // If editing an existing survey, populate the initial state with its data.
    if (survey) {
        emptySurvey.title = survey.title;
        emptySurvey.options = survey.options;
    }

    /**
     * Timeout duration for the failure alert popup.
     * @type {number}
     */
    const popUpTimeout: number = parseInt(process.env.REACT_APP_POPUP_TIMEOUT || "5000");

    const [showFailAlert, setShowFailAlert] = useState(false);
    const [state, setState] = useState<CreateSurvey>(emptySurvey);

    const navigate = useNavigate();

    // Effect to check if the user is logged in.
    useEffect(() => {
        UserService.isLoggedIn().then((isLoggedIn: boolean) => {
            if (!isLoggedIn) {
                navigate("/login");
            }
        });
    }, [navigate]);

    /**
     * Adds a recipe to the survey options.
     * @param {Recipe} recipe - The recipe to add.
     */
    const addRecipeToOptions = (recipe: Recipe) => {
        setState({ ...state, options: [...state.options, recipe] });
    };

    /**
     * Removes a recipe from the survey options.
     * @param {number} index - The index of the recipe to remove.
     */
    const removeFromOptions = (index: number) => {
        const newOptions = [...state.options];
        newOptions.splice(index, 1);
        setState({ ...state, options: newOptions });
    };

    /**
     * Validates the current state of the survey.
     * @returns {boolean} True if the state is valid, false otherwise.
     */
    const validate: () => boolean = () => {
        if (state.title.trim() === "") {
            return false;
        }
        return state.options.length >= 2;
    };

    /**
     * Saves the survey, either by creating a new one or updating an existing one.
     */
    const saveSurvey = () => {
        if (!validate()) {
            showPopUpFail();
            return;
        }
        if (survey) {
            const surveyUpdate: Survey = {
                id: survey.id,
                title: state.title,
                options: state.options.map((recipe: any) => `/recipes/${recipe.id}`),
                creator: survey.creator,
                creationDate: survey.creationDate,
                participants: survey.participants,
                recipeVote: survey.recipeVote
            };
            SurveyService.updateSurvey(surveyUpdate)
                .then((updatedSurvey) => {
                    window.location.assign(`/survey/view/${updatedSurvey.id}`);
                })
                .catch(() => {
                    showPopUpFail();
                });
            return;
        }
        SurveyService.createSurvey(state)
            .then((createdSurvey: Survey) => {
                window.location.assign(`/survey/view/${createdSurvey.id}`);
            })
            .catch(() => {
                showPopUpFail();
            });
    };

    /**
     * Shows the failure alert popup.
     */
    const showPopUpFail = () => {
        setShowFailAlert(true);
        setTimeout(() => setShowFailAlert(false), popUpTimeout);
    };

    return (
        <div>
            {showFailAlert && (
                <Alert
                    variant="danger"
                    onClose={() => setShowFailAlert(false)}
                    dismissible
                    className={"fixed-top"}
                >
                    Umfrage konnte nicht erstellt werden!
                </Alert>
            )}
            <LabelInput
                labelText="Title"
                initialValue={state.title}
                onChange={(val: string) => {
                    setState({ ...state, title: val });
                }}
            />
            <RecipeSearch onAdd={addRecipeToOptions} />
            <Heading headingText={"Rezepte"} />
            {state.options.map((recipe, index) => (
                <Stack direction={"horizontal"} key={index}>
                    <ImageArea origin={recipe.imgUri} />
                    <Link to={`/recipe/view/${recipe.id}`}>
                        <MyRecipeBar Recipe={recipe} />
                    </Link>
                    <Button variant="danger" className={"bi bi-x"} onClick={() => removeFromOptions(index)}></Button>
                </Stack>
            ))}
            <Stack direction="horizontal">
                <Button
                    variant="secondary"
                    onClick={() => {
                        if (survey) {
                            navigate(`/survey/`);
                        }
                        setState(emptySurvey);
                    }}
                    className="ms-auto"
                >
                    verwerfen
                </Button>
                <Button variant="primary" onClick={saveSurvey}>
                    speichern
                </Button>
            </Stack>
        </div>
    );
}

export default CreateSurveyView;