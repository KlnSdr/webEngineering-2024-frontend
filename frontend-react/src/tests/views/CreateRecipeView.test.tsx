import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import CreateSurveyView from "../../views/CreateSurveyView";
import {UserService} from "../../services/UserService";

jest.mock("../../services/UserService");

describe("CreateSurveyView Component", () => {
    it("redirects to login if user is not logged in", async () => {
        (UserService.isLoggedIn as jest.Mock).mockResolvedValue(false);
        render(
            <Router>
                <CreateSurveyView survey={null} />
            </Router>
        );
        await waitFor(() => expect(window.location.pathname).toBe("/login"));
    });

    it("renders correct when user is logged in", async () => {
        (UserService.isLoggedIn as jest.Mock).mockResolvedValue(true);
        const {asFragment} = render(
            <Router>
                <CreateSurveyView survey={null} />
            </Router>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});