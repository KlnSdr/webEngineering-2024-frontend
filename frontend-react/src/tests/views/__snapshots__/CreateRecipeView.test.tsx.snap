// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRecipeView Component renders correct when user is logged in and matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1
        class="headline-with-lines"
      >
        Title
      </h1>
      <input
        class="form-control"
        type="text"
        value=""
      />
    </div>
    <div>
      <h1
        class="headline-with-lines"
      >
        Bild Url
      </h1>
      <input
        class="form-control"
        type="text"
        value=""
      />
    </div>
    <h1
      class="headline-with-lines"
    >
      Zutaten
    </h1>
    <div>
      <button
        class="bi bi-plus btn btn-primary"
        data-testid="add-product"
        type="button"
      />
    </div>
    <h1
      class="headline-with-lines"
    >
      Zubereitung
    </h1>
    <textarea
      class="form-control"
      rows="10"
    />
    <div
      class="hstack"
    >
      <button
        class="ms-auto btn btn-secondary"
        type="button"
      >
        verwerfen
      </button>
      <button
        class="btn btn-primary"
        type="button"
      >
        speichern
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CreateRecipeView Component renders correctly when user is logged in and matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1
        class="headline-with-lines"
      >
        Title
      </h1>
      <input
        class="form-control"
        type="text"
        value=""
      />
    </div>
    <div>
      <h1
        class="headline-with-lines"
      >
        Bild Url
      </h1>
      <input
        class="form-control"
        type="text"
        value=""
      />
    </div>
    <h1
      class="headline-with-lines"
    >
      Zutaten
    </h1>
    <div>
      <button
        class="bi bi-plus btn btn-primary"
        data-testid="add-product"
        type="button"
      />
    </div>
    <h1
      class="headline-with-lines"
    >
      Zubereitung
    </h1>
    <textarea
      class="form-control"
      rows="10"
    />
    <div
      class="hstack"
    >
      <button
        class="ms-auto btn btn-secondary"
        type="button"
      >
        verwerfen
      </button>
      <button
        class="btn btn-primary"
        type="button"
      >
        speichern
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CreateSurveyView Component renders correct when user is logged in 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1
        class="headline-with-lines"
      >
        Title
      </h1>
      <input
        class="form-control"
        type="text"
        value=""
      />
    </div>
    <div>
      <div>
        <h1
          class="headline-with-lines"
        >
          Rezepte suchen
        </h1>
        <input
          class="form-control"
          type="text"
          value=""
        />
      </div>
    </div>
    <h1
      class="headline-with-lines"
    >
      Rezepte
    </h1>
    <div
      class="hstack"
    >
      <button
        class="ms-auto btn btn-secondary"
        type="button"
      >
        verwerfen
      </button>
      <button
        class="btn btn-primary"
        type="button"
      >
        speichern
      </button>
    </div>
  </div>
</DocumentFragment>
`;
